index.js

```diff
@@ -3,19 +3,11 @@
 var gutil     = require('gulp-util');
 var through   = require('through2');
 var chalk     = require('chalk');
+var hookStdio = require('hook-stdio');
 var Detector  = require('buddy.js/lib/detector');
 var reporters = require('buddy.js/lib/reporters');
 
-function hook_stdout(callback) {
-	process.stdout.write = (function(write) {
-		return function(string, encoding, fd) {
-			write.apply(process.stdout, arguments);
-			callback(string, encoding, fd);
-		};
-	})(process.stdout.write);
-}
-
-function checkOutput(output, reporter) {
+function buddySucceeded(output, reporter) {
```

Comment:
This function didn't make much sense as `checkOutput()`, it didn't imply what the return value would be. Much more readable now.